docs/01-app/02-building-your-application/07-configuring/05-mdx.mdx

```diff
@@ -248,6 +248,67 @@ export default function Page() {
 
 Navigating to the `/mdx-page` route should display your rendered MDX page.
 
+<AppOnly>
+
+### Using dynamic imports
+
+You can also dynamically import MDX in your pages.
+
+Create a new dynamic page within the `/app` directory and add MDX files wherever you'd like:
+
+```txt
+  my-project
+  ├── app
+  │   └── blog
+  │       └── [slug]
+  │           └── page.(tsx/js)
+  ├── markdown
+  │   ├── welcome.(mdx/md)
+      └── about.(mdx/md)
+  |── mdx-components.(tsx/js)
+  └── package.json
+```
+
+Dynamically import the MDX file inside the page and add a [`generateStaticParams`](/docs/app/api-reference/functions/generate-static-params) to statically prerender the pages with the MDX content:
+
+```tsx filename="app/blog/[slug]/page.tsx" switcher
+export default async function Page({
+  params,
+}: {
+  params: Promise<{ slug: string }>
+}) {
+  const slug = (await params).slug
+  const { default: Post } = await import(`@/markdown/${slug}.mdx`)
```

Comment:
needs a comment on how webpack dynamic importing works - I specifically was importing too broad:

`import('@${mdxPath}'` which was causing webpack to index every single file in the repo, and webpack was running out of memory. 

The static string of `@/markdown` is very important here, as is the static string of `.mdx` extension as otherwise it will find other files in that directory and can run into issues if it doesn't know how to import them. 

I have .mp4 files alongside my blog posts and it was hitting an issue where it didn't know how to handle mp4 files.

Reply:
Helpful, thank you!